backend/main.py

```
@@ -0,0 +1,550 @@
+    """
+    start_time = time.time()
+    llm_used = None
+
+    try:
+        question = query.question.lower().strip()
+
+        # Log the query (truncated for privacy)
+        logger.info(f"Processing query: {question[:50]}{'...' if len(question) > 50 else ''}")
+
+        # Define image-related keywords that indicate user wants illustrations
+        image_keywords = [
```

Ruff (0.12.2)

Local variable `image_keywords` is assigned to but never used

Remove assignment to unused variable `image_keywords`

(F841)